File: 0456-132-pattern/0456-132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        if len(nums) < 3:
            return False
        
        stack = []
        minPtr = -1
        for index in range(len(nums)-1, -1, -1):
            if minPtr != -1 and nums[minPtr] > nums[index]:
                return True
            
            while stack and nums[stack[-1]] < nums[index]:
                minPtr = stack.pop()
                
            stack.append(index)
            
            
        return False

File: 0456-132-pattern/0456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
         if (nums.size() < 3) {
        return false;
    }
    
    std::stack<int> stack;
    int minIndex = -1;
    
    for (int index = nums.size() - 1; index >= 0; index--) {
        if (minIndex != -1 && nums[index] < nums[minIndex]) {
            return true;
        }
        
        while (!stack.empty() && nums[stack.top()] < nums[index]) {
            minIndex = stack.top();
            stack.pop();
        }
        
        stack.push(index);
    }
    
    return false;
    }
};